### DISCUSSION TITLE
Why does Vercel "skip" commits and doesn't trigger a deploy? Project running behind (latest git commits not recognized)

### DISCUSSION THREAD

**Discussion by kyr0** [vercel/community#1018] [Help] [answered]
Hi,

I've a project setup with a public GitHub repo: https://github.com/kyr0/turbo-hybrid-astro-on-vercel 
The Vercel project is configured to deploy from the `main` branch.

When I push, Vercel triggers deploys, but only occasionally. It often "hangs" behind and doesn't recognize the most recent comments.
For every commit, where it worked as expected, you can see a comment attached to the commit:

<img width="1568" alt="Bildschirmfoto 2022-11-01 um 20 29 31" src="https://user-images.githubusercontent.com/454817/199321950-a12e17a7-2416-4135-ab12-fa497ca21272.png">

The Vercel App in GitHub is configured correctly and I already re-connected the integration to double-check. I also changed the branch name forth and back to main to double-check and started some Deploys manually, but when I do, Vercel still deploys from the old commit, not from the most recent one. 

Nothing helps, the last commit Vercel currently "sees" remains `f77c10a` at the time of writing this (4 commits behind).

This behaviour doesn't make sense to me. It looks like a background sync process is too slow or some cache invalidation doesn't work well. 

I can manually deploy as often as I want - deploys works, but it always picks the same old commit from the `main` branch until it finally finds the latest one after many minutes or even hours. And yes, of course I pushed to the remote; the difference between what Vercel sees as latest commit and what is on Github (main) can be verified by checking the repo commit history on Github and comparing with the screenshots below:

Here is the views from the Vercel UI:
<img width="1269" alt="Bildschirmfoto 2022-11-01 um 20 34 01" src="https://user-images.githubusercontent.com/454817/199322561-a885a2dd-6810-489d-9f5c-bf4154760b60.png">

<img width="1296" alt="Bildschirmfoto 2022-11-01 um 20 34 12" src="https://user-images.githubusercontent.com/454817/199322575-fbe61d15-9d84-4cfe-9dae-158915499f21.png">

It really doesn't want to pick the latest commits.

Do you have any ideas?

Thanks and best!

**Comment by amyegan** [comment, marked as answer]
Hi @kyr0! My first thought is you may have configured [Ignored Build Step](https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel) at some point, maybe with something like [`turbo-ignore`](https://vercel.com/changelog/intelligent-ignored-builds-using-turborepo) which could interfere with builds.

**Comment by kyr0** [comment]
@amyegan Thank you, Exactly! Unfortunately, I didn't see your answer and tried a few more times to debug the issue.
So I had to find it out the hard way xD Yes, indeed, `turbo-ignore` seems to be buggy or my turbo configuration is flawed (but it looks correct to my eyes tho). 

One thing I didn't notice before was this beautiful drop-down hiding the cancelled deployments:

<img width="210" alt="Bildschirmfoto 2022-11-01 um 21 21 56" src="https://user-images.githubusercontent.com/454817/199333683-8ec329c3-fca5-44a3-aab2-3f65a4ee146d.png">
<img width="231" alt="Bildschirmfoto 2022-11-01 um 21 22 02" src="https://user-images.githubusercontent.com/454817/199333693-058a91bd-0d86-460e-90c6-4fd80159102e.png">

Now with that one enabled, it's reveals, what happened (after clicking inspect deployment):
<img width="1245" alt="Bildschirmfoto 2022-11-01 um 21 22 11" src="https://user-images.githubusercontent.com/454817/199333828-1f424ce9-bcff-4c7e-ab9d-b30a34176382.png">

For now I disabled the ignore build step by configuration:
````
{
  "framework": "astro",
  "devCommand": "cd ../.. && yarn dev",
  "buildCommand": "cd ../.. && yarn build",
  "installCommand": "yarn --cwd=../.. install --frozen-lockfile",
  "ignoreCommand": "exit 1",
  "cleanUrls": true
}
````

**Reply by amyegan** [reply]
I'm glad builds are working for you now! I'll share this feedback with the team ❤️

**Reply by AlejandroSanchez90** [reply]
hi im triying to fix my deployment getting skipped aswell, where should i put that information, its my first time working with vercel, i started it as a create-next-app and dont have a vercel config file, how should i init it?

**Reply by Aliniroomand** [reply]
@amyegan I hope you best wishes by my heart...
I've been struggling and searching about 2 days for solving this problem and nothing found  but the only way that really worked was turbo-ignore...
thank you so much....

**Comment by tknickman** [comment]
@kyr0 glad you got things building! I wanted to add a little more info about `turbo-ignore` so hopefully you can still make use of it in the future. The way that `turbo-ignore` works is by finding the app at the **root** of your vercel project (for your project in this case, this is [apps/server](https://github.com/kyr0/turbo-hybrid-astro-on-vercel/tree/main/apps/server)). 

Then, for each build on Vercel, it checks to see if `server` itself, or _any of servers dependencies within the workspace_ (which in this case, is only [eslint-config-custom](https://github.com/kyr0/turbo-hybrid-astro-on-vercel/blob/main/apps/server/package.json#L31), [tsconfig](https://github.com/kyr0/turbo-hybrid-astro-on-vercel/blob/main/apps/server/package.json#L32), and [format](https://github.com/kyr0/turbo-hybrid-astro-on-vercel/blob/main/apps/server/package.json#L33)) have changed between the current commit, and the last successfully deployed commit for your current branch. 

If nothing is found - the build is ignored because turbo-ignore believe that the changes are not relevant to the app. 

For example, any changes in `README.md`, `public/` or `apps/static` would _not_ trigger a build. However changes in `apps/server`, or `packages/format` would. 

Hopefully that's helpful!


As an aside, we're working to make the docs here better so this is less "magic" and more understandable moving forward!

**Comment by perov93** [comment]
Quick solution, if someone is having problem with latest commit:

[https://vercel.com/changelog/manually-create-deployments-by-commit-or-branch-in-the-dashboard](url)

Hope it will help someone :)

**Reply by tella-john** [reply]
This works great, thanks! The underlying link is pointing to a different page though - here's a working link:
[https://vercel.com/changelog/manually-create-deployments-by-commit-or-branch-in-the-dashboard](https://vercel.com/changelog/manually-create-deployments-by-commit-or-branch-in-the-dashboard)
